repo/gopath.go

```diff
@@ -3,48 +3,15 @@ package repo
 import (
 	"fmt"
 
-	"github.com/shurcooL/Go-Package-Store/internal/util"
-	"github.com/shurcooL/go/exp/14"
-	"github.com/shurcooL/go/gists/gist7480523"
-	"github.com/shurcooL/go/gists/gist7802150"
+	"github.com/shurcooL/Go-Package-Store/pkg"
 )
 
 // GopathUpdater is an Updater that updates Go packages in local GOPATH workspaces.
-// Those packages are tracked in provided GoPackages.
-type GopathUpdater struct {
-	// GoPackages is a cached list of Go packages to work with.
-	GoPackages exp14.GoPackageList
-}
-
-func (u GopathUpdater) Update(importPathPattern string) error {
-	// TODO: This uses a legacy gist7802150 caching/cache-invalidation system. It's functional,
-	//       but poorly documented, has known flaws (it does not allow concurrent updates),
-	//       and very contributor-unfriendly (people don't like packages that have the word "gist" in
-	//       the import path, even if's not actually a gist; which is understandable, since it's basically
-	//       a package without a name that describes what it's for - that's something acceptable during
-	//       rapid prototyping, but not for the finished product). Need to redesign it and replace with
-	//       something better.
-	//
-	//       First step might be to simply drop the caching behavior and hope the user doesn't try
-	//       to manually refresh their browser page very often.
-
-	var updateErr = fmt.Errorf("import path pattern %q not found in GOPATH", importPathPattern)
-	gist7802150.MakeUpdated(u.GoPackages)
-	for _, goPackage := range u.GoPackages.List() {
-		if rootPath := util.GetRootPath(goPackage); rootPath != "" {
-			if gist7480523.GetRepoImportPathPattern(rootPath, goPackage.Bpkg.SrcRoot) == importPathPattern {
-
-				vcs := goPackage.Dir.Repo.RepoRoot.VCS
-				fmt.Printf("cd %s\n", rootPath)
-				fmt.Printf("%s %s", vcs.Cmd, vcs.DownloadCmd)
-				updateErr = vcs.Download(rootPath)
-
-				// Invalidate cache of the package's local revision, since it's expected to change after updating.
-				gist7802150.ExternallyUpdated(goPackage.Dir.Repo.VcsLocal.GetSources()[1].(gist7802150.DepNode2ManualI))
+type GopathUpdater struct{}
 
-				break
-			}
-		}
-	}
-	return updateErr
+func (GopathUpdater) Update(repo *pkg.Repo) error {
+	fmt.Printf("cd %s\n", repo.Path)
+	fmt.Printf("%s %s", repo.Cmd.Cmd, repo.Cmd.DownloadCmd)
```

Comment:
It's missing it on purpose.

The motivation is the following. Performing a `vcs.Download` operation can take a few seconds. During that time, I want the terminal output to reflect that it's in the process of doing a download, so it looks like this:

```
cd /path/to/repo
git pull --ff-only|
```

Where `|` indicates where the blinking caret is.

After the `vcs.Download` operation completes successfully, there will be a newline printed, followed by "Done.". The code for that is [here](https://github.com/shurcooL/Go-Package-Store/blob/b6082b0d2e39776793a01b27fc2734e4caf1a432/main.go#L117). Note its leading `\n` prefix there.